libindy/src/commands/crypto.rs

```diff
@@ -209,50 +288,368 @@ impl CryptoCommandExecutor {
+            .crypto_service
+            .crypto_box_seal_open(&my_key, &encrypted_msg)?;
 
         debug!("anonymous_decrypt <<< res: {:?}", res);
 
         Ok(res)
     }
 
     fn set_key_metadata(&self, wallet_handle: i32, verkey: &str, metadata: &str) -> Result<()> {
-        debug!("set_key_metadata >>> wallet_handle: {:?}, verkey: {:?}, metadata: {:?}", wallet_handle, verkey, metadata);
+        debug!(
+            "set_key_metadata >>> wallet_handle: {:?}, verkey: {:?}, metadata: {:?}",
+            wallet_handle, verkey, metadata
+        );
 
         self.crypto_service.validate_key(verkey)?;
 
-        let metadata = KeyMetadata {value: metadata.to_string()};
+        let metadata = KeyMetadata {
+            value: metadata.to_string(),
+        };
 
-        self.wallet_service.upsert_indy_object(wallet_handle, &verkey, &metadata)?;
+        self.wallet_service
+            .upsert_indy_object(wallet_handle, &verkey, &metadata)?;
 
         debug!("set_key_metadata <<<");
 
         Ok(())
     }
 
-    fn get_key_metadata(&self,
-                        wallet_handle: i32,
-                        verkey: &str) -> Result<String> {
-        debug!("get_key_metadata >>> wallet_handle: {:?}, verkey: {:?}", wallet_handle, verkey);
+    fn get_key_metadata(&self, wallet_handle: i32, verkey: &str) -> Result<String> {
+        debug!(
+            "get_key_metadata >>> wallet_handle: {:?}, verkey: {:?}",
+            wallet_handle, verkey
+        );
 
         self.crypto_service.validate_key(verkey)?;
 
-        let metadata = self.wallet_service.get_indy_object::<KeyMetadata>(wallet_handle, &verkey, &RecordOptions::id_value())?;
+        let metadata = self.wallet_service.get_indy_object::<KeyMetadata>(
+            wallet_handle,
+            &verkey,
+            &RecordOptions::id_value(),
+        )?;
 
         let res = metadata.value;
 
         debug!("get_key_metadata <<< res: {:?}", res);
 
         Ok(res)
     }
+
+    //TODO: Refactor pack to be more modular to version changes or crypto_scheme changes
+    //this match statement is super messy, but the easiest way to comply with current architecture
+    pub fn pack_msg(
+        &self,
+        message: Vec<u8>,
+        receivers: &str,
+        sender_vk: &str,
+        wallet_handle: i32,
+    ) -> Result<Vec<u8>> {
+
+        //parse receivers to structs
+        let receiver_list: Vec<String> = serde_json::from_str(receivers).map_err(|err| {
+            IndyError::CommonError(CommonError::InvalidStructure(format!(
+                "Failed to deserialize receiver list of keys {}",
+                err
+            )))
+        })?;
+
+        //break early and error out if no receivers keys are provided
+        if receiver_list.is_empty() {
+            return Err(IndyError::CommonError(CommonError::InvalidParam4(format!(
+                "No receiver keys found"
+            ))));
+        }
+        
+        match sender_vk.is_empty() {
```

Comment:
I suggest to replace this `match` by `if`

Reply:
Kept this as a match because the api now uses opt_c_str!() macro